mscore/editstyle.cpp

```diff
@@ -665,6 +673,39 @@ EditStyle::EditStyle(Score* s, QWidget* parent)
       cs->startCmd();
       }
 
+//---------------------------------------------------------
+//   showEvent
+//---------------------------------------------------------
+
+void EditStyle::showEvent(QShowEvent* ev)
+      {
+      if (isTooWide || isTooHigh) { // the default size is bigger than the screen
+            // MuseScore might have restored a smaller size than the default,
+            // which is the only size that the constructor sees. So the size
+            // must be checked here too in order to allow MuseScore to restore
+            // the user's last size.
+            QRect scr = QGuiApplication::primaryScreen()->geometry();
+            QRect dlg = this->frameGeometry();
+            if (dlg.width() > scr.width() || dlg.height() > scr.height()) {
+                  QRect inner = this->geometry(); // excludes borders and title bar
+                  int w = scr.width() - (dlg.width() - inner.width()); 
+                  int h = scr.height() - (dlg.height() - inner.height());
+                  if (inner.width() > w)
+                        inner.setWidth(w);
+                  if (inner.height() > h)
+                        inner.setHeight(h);
+                  this->setMinimumSize(inner.width() / 2, inner.height() / 2);
+                  this->setGeometry(inner);
+                  }
+
+            // Add scroll bars to pageStack
+            QScrollArea* scrollArea = new QScrollArea(splitter);
```

Comment:
This cannot be in the constructor, or weird sizing happens in the Headers, Footers page, possibly elsewhere.  This is the correct place for setting up QScrollArea.

Reply:
So every time this dialog gets hidden and shown again it will be granted an extra scroll area, won't it? What do you mean by weird sizing?

Reply:
Same if you put it in the constructor.  The constructor is called every time the dialog is opened via the menu.

Weird sizing is that the Header, Footer page is way too tall.  The text input boxes size taller.  I don't know why.

Reply:
`showEvent` gets called every time you minimize and restore MuseScore itself. Then one dialog object may get several scroll areas added, not sure whether this will have any visible effect though.

Reply:
OK, so then I can make the scroll area a member variable and initialize it only once.  Set it to zero in the .h file and check for a zero value when initializing.  I'll do that now.

Reply:
Though minimizing this dialog is a strange thing to do.  Minimizing any modal dialog is a weird thing to do.  Until you restore, your application is frozen.

Reply:
Oh, minimize and restore the  application, not the dialog.  Can you do that when the dialog is open?  Maybe just switching to and from another app is enough to cause the event to fire too.  I'll protect against this.

Reply:
I just wrote this code, then tested it and found that it is unnecessary.
There is no way to minimize MuseScore while the dialog is open.
The dialog box itself has no minimize button.
Switching to/from another application does not cause the showEvent() code to run.
There is no issue here.  The code does not need to change.

Reply:
> There is no way to minimize MuseScore while the dialog is open.

See the recording below. It is small due to size restrictions but it should probably be enough to demonstrate the issue.
Tested on Plasma 5 desktop, 800x600 resolution.
![screen](https://user-images.githubusercontent.com/6000747/56511900-7d84b800-6536-11e9-8065-3d3d71c28660.gif)

Reply:
I assume this is not Windows 10.  In Windows the dialog box has no minimize button.  I can recode what I had to prevent this.  I guess Windows has different rules than whatever OS this is for modal dialog boxes.  I'll let you know when this is handled.